rust/src/builtins/scriptlet_intercept/groupadd.rs

```diff
@@ -0,0 +1,150 @@
+//! CLI handler for intercepted `groupadd`.
+
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use anyhow::Result;
+use cap_std::fs::Dir;
+use clap::{Arg, Command};
+use std::io::Write;
+
+/// Entrypoint for (the rpm-ostree implementation of) `groupadd`.
+pub(crate) fn entrypoint(args: &[&str]) -> Result<()> {
+    fail::fail_point!("intercept_groupadd_ok", |_| Ok(()));
+
+    // This parses the same CLI surface as the real `groupadd`,
+    // but in the end we only extract the group name and (if
+    // present) the static GID.
+    let matches = cli_cmd().get_matches_from(args);
+    let gid: Option<u32> = match matches.value_of("gid") {
+        None => None,
+        Some(s) => Some(s.parse()?),
+    };
+    let groupname = matches
+        .value_of("groupname")
+        .expect("missing required groupname");
+
+    let rootdir = Dir::open_ambient_dir("/", cap_std::ambient_authority())?;
+    generate_sysusers_fragment(&rootdir, groupname, gid)?;
+
+    Ok(())
+}
+
+/// CLI parser, matches <https://linux.die.net/man/8/groupadd>.
+fn cli_cmd() -> Command<'static> {
+    let name = "groupadd";
+    Command::new(name)
+        .bin_name(name)
+        .about("create a new group")
+        .arg(Arg::new("force").short('f').long("force"))
+        .arg(
+            Arg::new("help")
+                .short('h')
+                .long("help")
+                .action(clap::ArgAction::Help),
+        )
+        .arg(Arg::new("gid").short('g').long("gid").takes_value(true))
+        .arg(Arg::new("key").short('K').long("key").takes_value(true))
+        .arg(Arg::new("allow_duplicates").short('o').long("non-unique"))
+        .arg(
+            Arg::new("password")
+                .short('p')
+                .long("password")
+                .takes_value(true),
+        )
+        .arg(Arg::new("system").short('r').long("system"))
```

Comment:
I think we should either:

- Error if this flag isn't passed
- No-op if this flag isn't passed

WDYT?

Reply:
Good idea, I think longer term we'd want to error out.
I'm not sure if we currently have scriptlets not using `-r`, so in the short term I'd just log a warning and proceed with the logic, so that we can adjust affected packages (if any).